cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -179,9 +229,11 @@ private DatasetMeta getFromMds(Id.DatasetInstance instance) throws Exception {
    * @throws DatasetTypeNotFoundException if the dataset type was not found
    * @throws Exception if something went wrong
    */
-  public void create(String namespaceId, String name, DatasetInstanceConfiguration props) throws Exception {
+  void create(String namespaceId, String name, DatasetInstanceConfiguration props) throws Exception {
     // Throws NamespaceNotFoundException if the namespace does not exist
     Id.Namespace namespace = ConversionHelpers.toNamespaceId(namespaceId);
+    Principal principal = SecurityRequestContext.toPrincipal();
+    authorizationEnforcementService.enforce(namespace.toEntityId(), principal, Action.WRITE);
     ensureNamespaceExists(namespace);
 
     Id.DatasetInstance newInstance = ConversionHelpers.toDatasetInstanceId(namespaceId, name);
```

Comment:
Standardized this according to https://wiki.cask.co/display/CE/Authorization+-+CDAP+3.5#Authorization-CDAP3.5-AuthorizationPolicies